### PR TITLE
Fix retrying logic

### PR DESCRIPTION
**Pull request by Fokko**
It was refreshing on each of the calls. I think the `before_sleep` is the correct hook.

### CODE REVIEW

File: tests/catalog/test_rest.py [hunk 1]
```diff
@@ -306,24 +306,39 @@ def test_list_namespace_with_parent_200(rest_mock: Mocker) -> None:
     ]
 
 
-def test_list_namespaces_419(rest_mock: Mocker) -> None:
+def test_list_namespaces_token_expired(rest_mock: Mocker) -> None:
     new_token = "new_jwt_token"
     new_header = dict(TEST_HEADERS)
     new_header["Authorization"] = f"Bearer {new_token}"
 
-    rest_mock.post(
```
Comment (Fokko) [inline tests/catalog/test_rest.py:314]: This `mock` was never called.

**Comment by Fokko** [comment]
cc @anupam-saini

**Review by anupam-saini — APPROVED** [review]
Thank you for catching this @Fokko

File: tests/catalog/test_rest.py (hunk 1 above)
Comment (anupam-saini) [reply tests/catalog/test_rest.py:314]: Perhaps this is why I didn't catch it while validating

**Review by anupam-saini — APPROVED** [review]

**Review by HonahX — APPROVED** [review]
LGTM! Thanks @Fokko for fixing this and thanks @anupam-saini for reviewing!

**Pull request merged**